updater-android/src/main/java/uk/os/unplugged/updater/android/Provider.java

```diff
@@ -0,0 +1,37 @@
+
+/*
+ * Copyright (C) 2016 Ordnance Survey
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.os.unplugged.updater.android;
+
+import java.io.File;
+
+/**
+ * Note: all references are assumed to have corresponding MD5 files.  For example, foo.bar is
+ * expected to have a corresponding foo.bar.md5 file.
+ */
+public interface Provider {
```

Comment:
Do we need a more expressive name for this? Not sure of Java code style, and perhaps the `package` namespace makes it clear, but if not, perhaps this is lacking some context.